aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,228 @@
+          "Cannot commit %s because Glue detected concurrent update", tableName());
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+      throw new AlreadyExistsException(e,
+          "Cannot commit %s because its Glue table already exists when trying to create one", tableName());
+    } catch (OperationTimeoutException | ResourceNumberLimitExceededException e) {
+      throw new CommitFailedException(e,
+          "Cannot commit %s because Glue operation or resource limit exceeded, " +
+              "please consider contact AWS to raise the limit", tableName());
+    } catch (Exception e) {
+      throw new RuntimeException("Unexpected exception during commit for " + tableName(), e);
+    } finally {
+      if (exceptionThrown) {
+        io().deleteFile(newMetadataLocation);
+      }
+    }
+  }
+
+  private void checkMetadataLocation(boolean isUpdate, Table glueTable, TableMetadata base) {
+    String glueMetadataLocation = isUpdate ? glueTable.parameters().get(METADATA_LOCATION_PROP) : null;
+    String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+    if (!Objects.equals(baseMetadataLocation, glueMetadataLocation)) {
+      throw new CommitFailedException(
+          "Cannot commit %s because base metadata location '%s' is not same as the current Glue location '%s'",
+          tableName(), baseMetadataLocation, glueMetadataLocation);
+    }
+  }
+
+  private Optional<Table> getGlueTable() {
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(databaseName)
+          .name(tableName)
+          .build());
+      return Optional.ofNullable(response.table());
+    } catch (EntityNotFoundException e) {
+      return Optional.empty();
+    }
+  }
+
+  private void updateParameters(Map<String, String> parameters, String newMetadataLocation) {
+    parameters.put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    parameters.put(METADATA_LOCATION_PROP, newMetadataLocation);
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
+      parameters.put(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation());
+    }
+  }
+
+  private void persistGlueTable(boolean isUpdate, Map<String, String> parameters) {
+    if (isUpdate) {
+      glue.updateTable(UpdateTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(databaseName)
+          .skipArchive(skipArchive)
+          .tableInput(TableInput.builder()
+              .name(tableName)
+              .parameters(parameters)
+              .build())
+          .build());
+    } else {
+      glue.createTable(CreateTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(databaseName)
+          .tableInput(TableInput.builder()
+              .name(tableName)
+              .owner(System.getProperty("user.name"))
```

Comment:
Is this required? Using `user.name` doesn't seem like a very accurate way to determine the user.

Reply:
Agree, I am basically following what Hive is doing here. There is no other meaningful information that can be put as input here because the Glue client cannot provide the caller information. I can also just remove the field if that is cleaner.